src/ucp/core/ucp_ep.c

```diff
@@ -1044,6 +1044,12 @@ void ucp_ep_disconnected(ucp_ep_h ep, int force)
 
     UCP_WORKER_THREAD_CS_CHECK_IS_BLOCKED(worker);
 
+    if (ucp_ep_is_cm_local_connected(ep)) {
```

Comment:
maybe move this if inside ucp_ep_cm_disconnect_cm_lane() ? 
and other places which call ucp_ep_cm_disconnect_cm_lane an also remove this if
maybe even rename it to `ucp_ep_cm_disconnect()`

Reply:
@yosefe but we still need the check for CM lane in
https://github.com/openucx/ucx/blob/bfca2be172b3a9903243eb1c71943f517d90a3f8/src/ucp/core/ucp_ep.c#L1134
https://github.com/openucx/ucx/blob/bfca2be172b3a9903243eb1c71943f517d90a3f8/src/ucp/core/ucp_ep.c#L1220

Reply:
decided not to do it